.github/workflows/test.yml

```diff
@@ -0,0 +1,42 @@
+name: Test
+
+on:
+  push:
+    branches: [ '*' ]
+  pull_request:
+    branches: [ '*' ]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest] # todo: windows-latest
+        node-version: [10.x, 12.x, 13.x, 14.x]
+
+    steps:
+    - name: Cancel Previous Runs
+      uses: styfle/cancel-workflow-action@0.4.0
+      if: ${{ github.event_name != 'pull_request' || github.event.pull_request.head.repo.id == github.event.pull_request.base.repo.id }}
```

Comment:
Wouldn't this prevent executing it for subsequent non-merge commits to the master branch? 🤔

Reply:
Please let me know what events you want to cancel.

Reply:
I would argue we should cancel previous workflow runs:
1. when new commit lands on the PR branch
2. when new commit lands on the master branch

What do you think? Does that make sense?

Reply:
Yeah, this line covers that (though it will cancel commits on any branch, and not just master). Merging a PR to master counts as a push to master (which is picked up as `github.event_name != 'pull_request'`) as well as just regular pushes to master.

Reply:
@MasterOdin Sorry for causing confusion I've somehow managed to interpret it inverted inside my head 🤦 
Yup, this seems ok 👍